pyof/v0x01/controller2switch/stats_request.py

```diff
@@ -44,29 +48,20 @@ def pack(self):
         Make `body` a binary pack before packing this object. Then, restore
         body.
         """
-        if self.body_type == StatsTypes.OFPST_PORT or \
-           self.body_type == StatsTypes.OFPST_FLOW or \
-           self.body_type == StatsTypes.OFPST_AGGREGATE:
-            backup = self.body
-            self.body = self.body.pack()
-            pack = super().pack()
-            self.body = backup
-            return pack
-        else:
+        if not self._types[self.body_type.value]:
             return super().pack()
 
+        backup = self.body
+        if hasattr(self.body, 'pack'):
+            self.body = self.body.pack()
+        pack = super().pack()
```

Comment:
If you are going to edit this commit, then if you can use a better name for the `pack` variable it would be nice! (I know it was already there). may be `packed` or `data` or something that does not cause a confusion between the method and the variable. ;)